airbyte-webapp-e2e-tests/cypress/commands/connectorBuilder.ts

```diff
@@ -66,4 +91,54 @@ export const assertMultiPageReadItems = () => {
 
   goToTestPage(3);
   testPanelContains('[]');
-};
\ No newline at end of file
+};
+
+const MAX_NUMBER_OF_PAGES = 5;
+export const assertMaxNumberOfPages = () => {
+  assertHasNumberOfPages(MAX_NUMBER_OF_PAGES)
+};
+
+export const assertHasNumberOfSlices = (numberOfSlices: number) => {
+  cy.get('[data-testid="tag-select-slice"] button').click();
+  cy.get('[data-testid="tag-select-slice"] li').should('have.length', numberOfSlices);
+};
+
+const MAX_NUMBER_OF_SLICES = 5;
+export const assertMaxNumberOfSlices = () => {
+  assertHasNumberOfSlices(MAX_NUMBER_OF_PAGES)
+};
+
+export const assertMaxNumberOfSlicesAndPages = () => {
+  for (var i = 0; i < MAX_NUMBER_OF_SLICES; i++) {
+    cy.get('[data-testid="tag-select-slice"] button').click();
+    cy.get('[data-testid="tag-select-slice"] li').contains("Slice " + i).click();
+    assertMaxNumberOfPages();
+  }
+};
+
+const SCHEMA =  ' {\n' +
+'   "$schema": "http://json-schema.org/schema#",\n' +
+'   "properties": {\n' +
+'     "name": {\n' +
+'       "type": "string"\n' +
+'     }\n' +
+'   },\n' +
+'   "type": "object"\n' +
+' }'
+export const assertSchema = () => {
+  cy.get('[data-testid="tag-tab-detected-schema"]').click();
+  cy.get('pre[class*="SchemaDiffView"]').contains(SCHEMA).should("exist");
```

Comment:
I feel this asserting is very precise and could break easily. For example, the JSON starts with a white space. If this change, this test will fail even though this might be perfectly fine

Reply:
You can parse the text as json and deep compare the objects:
```
// getDetectedSchemaElement is implemented in the page, see other comment
getDetectedSchemaElement().should(($el) => {
  expect($el.get(0).innerText).to.eq(SCHEMA_OBJECT)
})
```